### PR TITLE
Test for transaction security provider

### LINKED ISSUE #17810: Create e2e test for the transaction security provider

### PR DESCRIPTION
**Pull request by dzfjz**
Resolves #17810

### CODE REVIEW

**Review by PeterYinusa — DISMISSED** [review]

File: test/e2e/tests/security-provider.spec.js [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+const OPENSEA_URL =
+  'https://eos9d7dmfj.execute-api.us-east-1.amazonaws.com/metamask/validate';
+/**
+ * @param {import('mockttp').Mockttp} mockServer - The mock server.
+ */
+
+describe('Transaction security provider', function () {
+  let windowHandles;
+  // eslint-disable-next-line no-unused-vars
+  let extension;
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:14]: I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:107]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+        testApp = windowHandles[1];
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:109]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:152]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:193]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+
+        testApp = windowHandles[1];
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:195]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:236]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+
+        testApp = windowHandles[1];
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:238]: same here, I think we can remove this line as it's not needed anymore

File: test/e2e/tests/security-provider.spec.js [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+                'The security provider didn’t detect any known malicious activity, but it still may not be safe to continue.',
+              reason_header: 'Request may not be safe',
+            },
+          };
+        });
+        break;
+      case 'requestNotVerified':
+        await mockServer.forPost(OPENSEA_URL).thenCallback(() => {
+          return {
+            statusCode: 401,
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:59]: Here we should return a `500`response code, as we are mocking when the API server is down. See more here

https://github.com/MetaMask/metamask-extension/pull/17662#issuecomment-1437135605

File: test/e2e/tests/security-provider.spec.js [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+            json: {
+              flagAsDangerous: 3,
```
Comment (seaona) [inline test/e2e/tests/security-provider.spec.js:61]: the json body for this response should be all empty, as we are simulating when the API server is down, so there is no response on the body.

**Comment by seaona** [comment]
@dzfjz great job!! Nice to see it's working :D I added some small improvements but overall looks great!

File: test/e2e/tests/security-provider.spec.js (hunk 9 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 8 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 7 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 6 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 4 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js [hunk 11]
```diff
@@ -0,0 +1,255 @@
...
+        failOnConsoleError: false,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        await driver.openNewPage('http://127.0.0.1:8080/');
+        windowHandles = await driver.getAllWindowHandles();
+        extension = windowHandles[0];
```
[the remark this answers is not in the document]
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 3 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 2 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js [hunk 12]
```diff
@@ -0,0 +1,255 @@
...
+/**
+ * @param {import('mockttp').Mockttp} mockServer - The mock server.
+ */
+
+describe('Transaction security provider', function () {
+  let windowHandles;
+  // eslint-disable-next-line no-unused-vars
+  let extension;
+  // eslint-disable-next-line no-unused-vars
+  let testApp;
```
[the remark this answers is not in the document]
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

File: test/e2e/tests/security-provider.spec.js (hunk 1 above)
Comment (dzfjz) [reply test/e2e/tests/security-provider.spec.js]: Done

**Review by PeterYinusa — APPROVED** [review]

**Review by seaona — APPROVED** [review]

**Pull request merged**